pennylane/ops/qubit/qchem_ops.py

```diff
@@ -114,12 +124,39 @@ def compute_matrix(phi):  # pylint: disable=arguments-differ
                 [ 0.0000,  0.2474,  0.9689,  0.0000],
                 [ 0.0000,  0.0000,  0.0000,  1.0000]])
         """
-        c = qml.math.cos(phi / 2)
+        interface = qml.math.get_interface(phi)
+        if interface == "tensorflow":
+            c = qml.math.cos(phi / 2)
+            s = qml.math.sin(phi / 2)
+            ones = qml.math.ones_like(phi)
+            zeros = qml.math.zeros_like(phi)
+            rows = [
+                [ones, zeros, zeros, zeros],
```

Comment:
Why not just  using something like `qml.math.diag([1, c, c, 1])`?

Reply:
At this point, we do not differentiate between broadcasted (i.e. a vector) `phi` or scalar `phi`. That means, that the output matrix might be 3-dimensional and unfortunately `np.diag` or `qml.math.diag` do not support this use case (passing a 2d-array to `diag` extracts the diagonal, instead of broadcasting the `1d->2d` functionality across one of the axes :/ )